run_infrastructure.py

```diff
@@ -0,0 +1,241 @@
+"""
+This file invokes a command used to setup infrastructure for use in testing Fidesops
+and related workflows.
+"""
+import argparse
+import os
+import sys
+from typing import (
+    List,
+)
+
+DOCKER_WAIT = 15
+DOCKERFILE_DATASTORES = [
+    "postgres",
+    "mysql",
+    "mongodb",
+    "mssql",
+]
+EXTERNAL_DATASTORE_CONFIG = {
+    "snowflake": ["SNOWFLAKE_TEST_URI"],
+    "redshift": ["REDSHIFT_TEST_URI", "REDSHIFT_TEST_DB_SCHEMA"],
+}
+EXTERNAL_DATASTORES = list(EXTERNAL_DATASTORE_CONFIG.keys())
+IMAGE_NAME = "fidesops"
+
+
+def run_infrastructure(
+    datastores: List[str] = [],  # Which infra should we create? If empty, we create all
+    open_shell: bool = False,  # Should we open a bash shell?
+    pytest_path: str = "",  # Which subset of tests should we run?
+    run_application: bool = False,  # Should we run the Fidesops webserver?
+    run_quickstart: bool = False,  # Should we run the quickstart command?
+    run_tests: bool = False,  # Should we run the tests after creating the infra?
+) -> None:
+    """
+    - Create a Docker Compose file path for all datastores specified in `datastores`.
+    - Defaults to creating infrastructure for all datastores in `DOCKERFILE_DATASTORES` if none
+    are provided.
+    - Optionally runs integration tests against those datastores from the container identified
+    with `IMAGE_NAME`.
+    """
+
+    if len(datastores) == 0:
+        os.system(
+            f'echo "no datastores specified, configuring infrastructure for all datastores"'
+        )
+        datastores = DOCKERFILE_DATASTORES + EXTERNAL_DATASTORES
+    else:
+        os.system(f'echo "datastores specified: {", ".join(datastores)}"')
+
+    # De-duplicate datastores
+    datastores = set(datastores)
+
+    # Configure docker-compose path
+    path: str = get_path_for_datastores(datastores)
+
+    os.system(f'echo "infrastructure path: {path}"')
+    os.system(f"docker-compose {path} build")
+    os.system(f'echo "sleeping for: {DOCKER_WAIT} while infrastructure loads"')
+    os.system(f"sleep {DOCKER_WAIT}")
+
+    seed_initial_data(
+        datastores,
+        path,
+        base_image=IMAGE_NAME,
+    )
+
+    if open_shell:
+        return _open_shell(path, IMAGE_NAME)
+
+    elif run_application:
+        return _run_application(path)
+
+    elif run_quickstart:
+        return _run_quickstart(path, IMAGE_NAME)
+
+    elif run_tests:
+        # Now run the tests
+        return _run_tests(
+            datastores,
+            docker_compose_path=path,
+            pytest_path=pytest_path,
+        )
+
+
+def seed_initial_data(
+    datastores: List[str],
+    path: str,
+    base_image: str,
+) -> None:
+    """
+    Seed the datastores with initial data as defined in the file at `setup_path`
+    """
+    for datastore in datastores:
+        if datastore in DOCKERFILE_DATASTORES:
+            setup_path = f"tests/integration_tests/{datastore}_setup.py"
```

Comment:
This does seem like a better pattern than the (convenient, but dangerous) SQL script that the postgres/mysql/etc. images use to auto-populate their code. That sounds like a separate feature request, though, to "Migrate SQL setup scripts to Python scripts for Postgres, MySQL, etc.", and would be a pretty good issue for new contributors

Reply:
Ultimately, with the next ticket, we'll move only schema creation to these files where required. The test data will always be added through Pytest fixtures so that we can run access and erasure tests together and not through separate markers